### PR TITLE
Sort a work's files in alphabetical order by default

### LINKED ISSUE #1044: Files should be alphabetically sorted
# Feature Request

When viewing a work, the list of files in the work should be sorted alphabetically by filename. Currently they are sorted by the order they were uploaded (I think).

### PR DESCRIPTION
**Pull request by nicktiberi**
Fixes #1044.

### CODE REVIEW

File: spec/models/work_version_spec.rb [hunk 1]
```diff
@@ -59,6 +59,14 @@
 
       it_behaves_like 'a resource with orderable creators'
     end
+
+    describe '#file_version_memberships' do
+      let(:resource) { create(:work_version, :with_files, file_count: 2) }
+
+      it 'orders the files by title in ascending order' do
+        expect(resource.file_version_memberships.map(&:title)).to eq(['image-1.png', 'image-2.png'])
```
Comment (awead) [inline spec/models/work_version_spec.rb]: You need to rename the files and verify that they get sorted correctly. Something like:
``` ruby
resource.file_version_memberships[0].update(title: 'image-4.png')
resource.file_version_memberships[1].update(title: 'image-3.png')
```
ought to do the trick.

**Review by awead — CHANGES_REQUESTED** [review]

File: app/models/file_version_membership.rb [hunk 2]
```diff
@@ -4,6 +4,8 @@ class FileVersionMembership < ApplicationRecord
   belongs_to :work_version
   belongs_to :file_resource
 
+  default_scope { order(title: :asc) }
```
Comment (awead) [inline app/models/file_version_membership.rb:7]: ah nice! I forgot about that....

**Comment by nicktiberi** [comment]
@awead - I have one failing test and am wondering if you have any idea why it's failing. 

I ran `spec/services/file_version_membership_diff_spec.rb` locally and everything passed, but it seems to still be failing in CI.

Based on expected/got, it seems like the timestamps are different, and that may be what's causing it to fail. But I'm not sure why the timestamps are different.

```
Failures:

  1) FileVersionMembershipDiff when a file has been renamed is expected to eq {:added=>[], :deleted=>[], :renamed=>[[#<FileVersionMembership id: 537, work_version_id: 1014, file_r...ted_at: "2021-06-01 11:01:39.599640000 -0400", updated_at: "2021-06-01 11:01:39.599640000 -0400">]]}
     Failure/Error: it { is_expected.to eq(renamed: [[original_second_image, updated_second_image]], added: [], deleted: []) }
     
       expected: {:added=>[], :deleted=>[], :renamed=>[[#<FileVersionMembership id: 537, work_version_id: 1014, file_r...ted_at: "2021-06-01 11:01:39.599640000 -0400", updated_at: "2021-06-01 11:01:39.599640000 -0400">]]}
            got: {:added=>[], :deleted=>[], :renamed=>[[#<FileVersionMembership id: 537, work_version_id: 1014, file_r...ted_at: "2021-06-01 11:01:39.602285000 -0400", updated_at: "2021-06-01 11:01:39.611688000 -0400">]]}
     
       (compared using ==)
     
       Diff:
       @@ -1,4 +1,4 @@
        :added => [],
        :deleted => [],
       -:renamed => [[#<FileVersionMembership id: 537, work_version_id: 1014, file_resource_id: 546, title: "image-554.png", created_at: "2021-06-01 11:01:39.566987000 -0400", updated_at: "2021-06-01 11:01:39.566987000 -0400">, #<FileVersionMembership id: 538, work_version_id: 1015, file_resource_id: 545, title: "image-553.png", created_at: "2021-06-01 11:01:39.599640000 -0400", updated_at: "2021-06-01 11:01:39.599640000 -0400">]],
       +:renamed => [[#<FileVersionMembership id: 537, work_version_id: 1014, file_resource_id: 546, title: "image-554.png", created_at: "2021-06-01 11:01:39.566987000 -0400", updated_at: "2021-06-01 11:01:39.566987000 -0400">, #<FileVersionMembership id: 539, work_version_id: 1015, file_resource_id: 546, title: "Image 2.png", created_at: "2021-06-01 11:01:39.602285000 -0400", updated_at: "2021-06-01 11:01:39.611688000 -0400">]],
       
     # ./spec/services/file_version_membership_diff_spec.rb:22:in `block (3 levels) in <top (required)>'

Finished in 7 minutes 10 seconds (files took 11.19 seconds to load)
2092 examples, 1 failure, 20 pending

Failed examples:

rspec ./spec/services/file_version_membership_diff_spec.rb:22
```

**Comment by nicktiberi** [comment]
@awead - I think codeclimate is just stuck. This should be good for re-review now.

File: spec/models/work_version_spec.rb [hunk 3]
```diff
@@ -59,6 +59,24 @@
 
       it_behaves_like 'a resource with orderable creators'
     end
+
+    describe '#file_version_memberships' do
+      let(:resource) { create(:work_version, :with_files, file_count: 2) }
+
+      it 'orders the files by title in ascending order' do
+        n1 = (resource.file_version_memberships[0].title.match /image-(\d+).png/)[1].to_i
+        n2 = n1 + 1
```
Comment (awead) [inline spec/models/work_version_spec.rb:68]: clever!

**Review by awead — APPROVED** [review]

**Pull request merged**